docs/sphinx/source/introexamples.rst

```diff
@@ -0,0 +1,246 @@
+        pressure = pvlib.atmosphere.alt2pres(altitude)
+        am_abs = pvlib.atmosphere.get_absolute_airmass(airmass, pressure)
+        tl = pvlib.clearsky.lookup_linke_turbidity(times, latitude, longitude)
+        cs = pvlib.clearsky.ineichen(solpos['apparent_zenith'], am_abs, tl,
+                                     dni_extra=dni_extra, altitude=altitude)
+        aoi = pvlib.irradiance.aoi(system['surface_tilt'], system['surface_azimuth'],
+                                   solpos['apparent_zenith'], solpos['azimuth'])
+        total_irrad = pvlib.irradiance.get_total_irradiance(system['surface_tilt'],
+                                                            system['surface_azimuth'],
+                                                            solpos['apparent_zenith'],
+                                                            solpos['azimuth'],
+                                                            cs['dni'], cs['ghi'], cs['dhi'],
+                                                            dni_extra=dni_extra,
+                                                            model='haydavies')
+        temps = pvlib.pvsystem.sapm_celltemp(total_irrad['poa_global'],
+                                             wind_speed, temp_air)
+        effective_irradiance = pvlib.pvsystem.sapm_effective_irradiance(
+            total_irrad['poa_direct'], total_irrad['poa_diffuse'],
+            am_abs, aoi, module)
+        dc = pvlib.pvsystem.sapm(effective_irradiance, temps['temp_cell'], module)
+        ac = pvlib.pvsystem.snlinverter(dc['v_mp'], dc['p_mp'], inverter)
+        annual_energy = ac.sum()
+        energies[name] = annual_energy
+
+    energies = pd.Series(energies)
+
+    # based on the parameters specified above, these are in W*hrs
+    print(energies.round(0))
+
+    energies.plot(kind='bar', rot=0)
+    @savefig proc-energies.png width=6in
+    plt.ylabel('Yearly energy yield (W hr)')
+    @suppress
+    plt.close();
+
+
+.. _object-oriented:
+
+Object oriented (Location, PVSystem, ModelChain)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The first object oriented paradigm uses a model where a
+:py:class:`~pvlib.pvsystem.PVSystem` object represents an assembled
+collection of modules, inverters, etc., a
+:py:class:`~pvlib.location.Location` object represents a particular
+place on the planet, and a :py:class:`~pvlib.modelchain.ModelChain`
+object describes the modeling chain used to calculate PV output at that
+Location. This can be a useful paradigm if you prefer to think about the
+PV system and its location as separate concepts or if you develop your
+own ModelChain subclasses. It can also be helpful if you make extensive
+use of Location-specific methods for other calculations. pvlib-python
+also includes a :py:class:`~pvlib.tracking.SingleAxisTracker` class that
+is a subclass of :py:class:`~pvlib.pvsystem.PVSystem`.
+
+The following code demonstrates how to use
```

Comment:
I'd add some text here about the defaults in `ModelChain` because our example provides the absolute minimum input.  Perhaps

`ModelChain` objects provide convenience methods that can provide default selections for models and can also fill necessary input data with modeled data. In our example below, we use convenience methods.  For example, no irradiance data is provided as input, so the `ModelChain` object substitutes irradiance from a clear-sky model via the `prepare_inputs` method.  Also, no irradiance transposition model is specified (kwarg `transposition` for `ModelChain`) so the `ModelChain` defaults to the `haydavies` model.  In this example, `ModelChain` infers the DC power model from the `module` provided by examining the parameters defined for `module`.

Reply:
good idea, thanks.